src/main/java/fr/xephi/authme/process/AsyncUserScheduler.java

```diff
@@ -0,0 +1,47 @@
+package fr.xephi.authme.process;
+
+import com.jano7.executor.KeySequentialRunner;
+import fr.xephi.authme.service.BukkitService;
+import org.bukkit.entity.Player;
+
+import javax.inject.Inject;
+
+/**
+ * Handles the queue of async tasks on a per-player basis.
+ */
+public class AsyncUserScheduler {
+
+    @Inject
+    private BukkitService bukkitService;
+
+    private KeySequentialRunner<String> asyncUserScheduler;
+
+    AsyncUserScheduler() {
+        this.asyncUserScheduler = new KeySequentialRunner<>(command -> bukkitService.runTaskAsynchronously(command));
```

Comment:
Maybe I'm missing here something, but this appears just submits a task to be run async and returns immediately. The tasks will still run in concurrently.

Reply:
Hmm, what do you mean exactly? I implemented a custom Executor which schedules async tasks using the bukkit scheduler.

Reply:
I thought the idea is to run the player specific code sequential for the same player. `runTaskAsynchronously` submits the task to Bukkit and then returns. This means the queue for this `UserScheduler` is empty. Another task can be submitted while the older is still running, because of the empty queue.

Reply:
Oh, I'm soo stupid lol, guess we have to use a custom executor at this point.

Reply:
Something isn't quite right anyway, this is the library I used: https://github.com/jano7/executor
From what I understand it won't wait for the task to complete, right?

Reply:
The ReadMe sounds like exactly what you want. Processing for the equal key is performed sequentially. 

I might just realized that I'm completely wrong. We are already creating a custom executor, because of the lambda call. So in theory there should be no issue.

Reply:
yep but what does track when a task is completed?

Reply:
Ok figured it out lol, any keyed runner keeps a runnable queue and runs tasks sequentially. Works as expected :)

Reply:
> yep but what does track when a task is completed?

They seem to wrap the runnable to track it.